### PR TITLE
init: Initial commit of the client commands

### PR DESCRIPTION
**Pull request by mattyw**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+	"github.com/juju/errors"
+	rterms "github.com/juju/romulus/api/terms"
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+
+	"github.com/juju/terms-client/api/wireformat"
+)
+
+const baseURL = "https://api.jujucharms.com/terms"
+
+// Client reprensents the interface of the terms service.
```
Comment (cmars) [inline api/api.go]: represents (spelling)
Also, probably better to say something like "Client represents the terms service API."

File: api/api.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+type Client interface {
+	SaveTerm(owner, name, content string) (string, error)
```
Comment (cmars) [inline api/api.go]: Would be nice to document these here I think. I don't thing the unexported `client` type will show godocs for its methods.

File: api/wireformat/entities.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+// TermIDResponse contains just the termID
+type TermIDResponse struct {
+	TermID string `json:"term-id"`
+}
+
+func (t *Term) termID() string {
+	return fmt.Sprintf("%s/%s/%d", t.Owner, t.Name, t.Revision)
+}
+
+// Terms stores a slice of terms.
```
Comment (cmars) [inline api/wireformat/entities.go]: Terms makes a slice of Terms sortable.

File: api/wireformat/entities.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+type Terms []Term
+
+func (terms Terms) Len() int {
```
Comment (cmars) [inline api/wireformat/entities.go]: Len implements sort.Interface, etc.

File: api/wireformat/entities.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+// AgreementRequest holds the parameters for creating a new
+// user agreement to a specific revision of terms.
+type AgreementRequest struct {
+	TermOwner    string `json:"termowner"`
+	TermName     string `json:"termname"`
+	TermRevision int    `json:"termrevision"`
+}
+
+type Agreements struct {
```
Comment (cmars) [inline api/wireformat/entities.go]: godoc please

**Comment by cmars** [comment]
Just nitpicks, but I'd like this to read well on godoc.org.

File: api/api.go [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+	Do(*http.Request) (*http.Response, error)
+	DoWithBody(req *http.Request, body io.ReadSeeker) (*http.Response, error)
+}
+
+// ClientOption defines a function which configures a Client.
+type ClientOption func(h *client)
+
+// HTTPClient returns a function that sets the http client used by the API
+// (e.g. if we want to use TLS).
+func HTTPClient(c httpClient) func(h *client) {
```
Comment (alesstimec) [inline api/api.go]: func HTTPClient(c httpClient) ClientOption {

File: api/api.go [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+	return func(h *client) {
+		h.bclient = c
+	}
+}
+
+// ServiceURL returns a function that sets the terms service URL used
+// by the API.
+func ServiceURL(serviceURL string) func(h *client) {
```
Comment (alesstimec) [inline api/api.go]: func ServiceURL(serviceURL string) ClientOption) {

File: api/wireformat/entities.go [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+	if err != nil {
+		return errors.Trace(err)
+	}
+	*t = TimeRFC3339(t0)
+	return nil
+}
+
+// DebugStatusResponse contains results of various checks that
+// form the status of the terms service.
+type DebugStatusResponse struct {
```
Comment (alesstimec) [inline api/wireformat/entities.go]: i don't think this type is used anywhere

File: api/wireformat/entities.go [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+	Checks map[string]CheckResult `json:"checks"`
+}
+
+// CheckResult holds the result of a single status check.
+type CheckResult struct {
```
Comment (alesstimec) [inline api/wireformat/entities.go]: same for this type

**Comment by alesstimec** [comment]
a few nitpicks, then LGTM

**Comment by alesstimec** [comment]
LGTM

**Comment by cmars** [comment]
LGTM

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Comment by mattyw** [comment]
$$merge$$

**Pull request merged**